tests/scanner/scanners/service_account_key_test.py

```diff
@@ -0,0 +1,165 @@
+""" % (UN_ROTATED_KEY_NAME, UN_ROTATED_KEY_TIME, VALID_BEFORE_TIME)
+
+RESOURCE_DATA = "{'key_id': u'999999999999999999999', " \
+                "'key_algorithm': u'KEY_ALG_RSA_2048', " \
+                "'valid_after_time': u'%s', " \
+                "'valid_before_time': u'%s', " \
+                "'full_name': u'organization/12345/project/foo/serviceaccount" \
+                "/123456789012345678901/serviceaccount_key" \
+                "/999999999999999999999/'}" % (UN_ROTATED_KEY_TIME,
+                                               VALID_BEFORE_TIME)
+
+EXPECTED_VIOLATION = {'rule_name': 'Service account keys not rotated (older than 100 days)',
+                      'resource_data': RESOURCE_DATA,
+                      'full_name': u'organization/12345/project/foo/serviceaccount/123456789012345678901/serviceaccount_key/999999999999999999999/',
+                      'resource_id': u'test-service-account@developer.gserviceaccount.com',
+                      'rule_index': 0,
+                      'violation_type': 'SERVICE_ACCOUNT_KEY_VIOLATION',
+                      'violation_data': {
+                          'service_account_name': u'Service Account Key Age Test Service Account',
+                          'service_account_id': u'test-service-account@developer.gserviceaccount.com',
+                          'violation_reason': u'Key ID 999999999999999999999 not rotated since %s.' % UN_ROTATED_KEY_TIME,
+                          'key_id': u'999999999999999999999',
+                          'project_id': u'foo',
+                          'key_created_time': u'%s' % UN_ROTATED_KEY_TIME},
+                      'resource_type': 'serviceaccount_key'
+                      }
+
+
+class FakeServiceConfig(object):
+
+    def __init__(self):
+        engine = create_test_engine()
+        self.model_manager = ModelManager(engine)
+
+
+# pylint: disable=bad-indentation
+class ServiceAccountKeyScannerTest(unittest_utils.ForsetiTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.service_config = FakeServiceConfig()
+        cls.model_name = cls.service_config.model_manager.create(
+            name='service-account-key-scanner-test')
+
+        scoped_session, data_access = (
+            cls.service_config.model_manager.get(cls.model_name))
+
+        # Add organization and project to model.
+        with scoped_session as session:
+            organization = data_access.add_resource_by_name(
+                session, 'organization/12345', '', True)
+            project = data_access.add_resource(session, 'project/foo',
+                                               organization)
+            service_account = data_access.add_resource(
+                session, 'serviceaccount/%s' % SERVICE_ACCOUNT_ID,
+                project)
+            service_account.data = SERVICE_ACCOUNT_DATA
+
+            # Add both, the rotated and un-rotated keys to the above
+            sc = data_access.add_resource(
```

Comment:
What is the abbreviation `sc`?  Can we be a bit more verbose here so that it's more easy to read?

Reply:
Changed to "rotated_key" and "un_rotated_key", so it makes more sense.